package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -818,6 +818,8 @@ dnsmasq_ipset_add() {
 		domains="$domains/$1"
 	}
 
+	config_get table "$cfg" table 'fw4'
```

Comment:
@ldir-EDB0  I'm not sure I understand this code correctly, but can we not get the type and table name from the nftset option definition?
What's the suggested syntax for the nftset definition in the config file? Can we keep it the same as the dnsmasq definition, like: `   option nftset '/domain.com/<family>#<type>#<table>#<set>'`?

Reply:
The nftset option lives in /etc/config/firewall, not in /etc/config/dhcp.
Similarly there's no guarantee the set is defined or under control of fw4, it might be in a different table controlled by a 3rd party application.  - it may not even exist at dnsmasq startup time, this is also the reason why I did the ip version heuristic based on the set name.

Reply:
@ldir-EDB0 thank you for your prompt reply. I believe I have unintentionally created a confusion and I should have phrased things better. I'm not talking about defining a new nftset, which, similarly to defining a new ipset is done thru the firewall config.

I'm talking about the option to let the dnsmasq resolve the domain and add elements to the nftset. The similar support for ipset required entries in `/etc/config/dhcp`: https://openwrt.org/docs/guide-user/base-system/dhcp#all_options.

If the new OpenWrt dnsmasq-full package supports resolving domains and adding their IP addresses to the specified nft set, I believe this should also be configured in `/etc/config/dhcp` and that config entry should be similar to what upstream dnsmasq package requires, specifically: ip_family, table_type, table_name, set_name.

That's why I'd suggest that the entries for these should be in `/etc/config/dhcp` and should be:
```
list nftset '/domain.com/<ip_family>#<table_type>#<table_name>#<set_name>'
```

This way, the dnsmasq init can parse the dhcp config and obtain all the required information for the nft set from the config entry.

I'm directly affected by this PR, as I maintain two packages which would benefit from the nft set support in dnsmasq -- simple-adblock and vpn-policy-routing. The latter wil be replaced by the new package `pbr` which is now nft-compatible, but I'm waiting for the nft set support in dnsmasq before I submit a PR to replace the old, iptables-only package.

Reply:
The new dnsmasq init script will still read the existing `ipset` config entries in `/etc/config/dhcp` for backward compatibility. The same `ipset` entries can be used by `fw3` or `fw4`.

Reply:
I believe @jow- has previously said that the ipset entries from the firewall config will be created as nft sets by fw4.

Anyways, if dnsmasq-full supports both ipset and nft sets, it should parse both ipset and nftset entries from its config and add IP addresses to ipset and nft set respectively.

Reply:
and this is exactly why I don't want to get involved with dnsmasq bumps.

Reply:
Since commit acb336235cc640980ea549655b0702daf9e628f8 the preferred dnsmasq ipset configuration is a separate section in the dhcp config. Both of dnsmasq’s native options `ipset=` and `nftset=` will be populated from the same ipset definition in `/etc/config/dhcp`. There is no dedicated nftset configuration in the uci dhcp config.

Reply:
add example config which hopefully makes things clearer

Reply:
> Since commit [acb3362](https://github.com/openwrt/openwrt/commit/acb336235cc640980ea549655b0702daf9e628f8) the preferred dnsmasq ipset configuration is a separate section in the dhcp config. Both of dnsmasq’s native options `ipset=` and `nftset=` will be populated from the same ipset definition in `/etc/config/dhcp`. There is no dedicated nftset configuration in the uci dhcp config.

Could you please help me understand how this is supposed to work by providing sample configs for the following scenarios for domain google.com when the user wants the IPs from resolved domain to be added to:

1. the ipsets named `set_4_cfg000` and `set_6_cfg000`
2. the nft set `set_4_cfg000` of IPv4 family in fw4 table and the nft set `set_6_cfg000` of IPv6 family in the same table
3. the nft set `nftset_4_cfg001` of IPv4 family in the table `custom4` of type `ip` and the nft set `nftset_6_cfg001` of IPv6 family in the table `custom6` of type `ip6`
4. the nft set `nftset_4_cfg002` of IPv4 family in the table `custom4` of type `inet` and the nft set `nftset_6_cfg002` in the same table

Thanks!

Reply:
Scenarios 1,2,4 would be naming variations on this sample config for `/etc/config/dhcp`:
```
config ipset
    list name 'nftset_4_cfg002'
    list name 'nftset_6_cfg002'
    list domain 'google.com'
    option table 'custom4' # optional, defaults to fw4
```
Scenario 3 would break because everything has been assumed to be an `inet` table so far, since fw4 uses an `inet` table. But if you're already into making custom tables, you might not be averse to adding custom config to `dnsmasq.conf.d`.

Reply:
I appreciate your reply @dave14305 even tho Kevin has closed this PR.

So how does a user make dnsmasq add IPs to an ipset vs nft set if configuration is the same?